LiteDB/Utils/Extensions/WaitHandleExtensions.cs

```diff
@@ -0,0 +1,33 @@
+﻿namespace LiteDB.Utils.Extensions
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    internal static class WaitHandleExtensions
+    {
+        public static async Task<bool> WaitAsync(this WaitHandle handle)
+        {
+            var tcs = new TaskCompletionSource<bool>();
+
+            using (new ThreadPoolRegistration(handle, tcs))
+            {
+                return await tcs.Task.ConfigureAwait(false);
+            }
+        }
+
+        private sealed class ThreadPoolRegistration : IDisposable
```

Comment:
If we change this to be a `struct` does the code work as expected? If so, let's change it and implement the interface implicit, that way we don't need to box it.

Changing this to a `struct` will avoid allocations if the task is already completed

```suggestion
        private readonly struct ThreadPoolRegistration : IDisposable
```